extensions/notebook/src/jupyter/jupyterKernel.ts

```diff
@@ -87,6 +87,8 @@ export class JupyterKernel implements nb.IKernel {
 	}
 
 	requestExecute(content: nb.IExecuteRequest, disposeOnDone?: boolean): nb.IFuture {
+		content.code = Array.isArray(content.code) ? content.code.join('') : content.code;
+		content.code = content.code.split(/[\r\n]+/gm).join('\n');
```

Comment:
Can we use replace? It's more efficient. Also cover multiple '\r' just in case. Jupyterlab output response processing for reference:

```ts
  function fixCarriageReturn(txt: string): string {
    txt = txt.replace(/\r+\n/gm, '\n'); // \r followed by \n --> newline
    while (txt.search(/\r[^$]/g) > -1) {
      const base = txt.match(/^(.*)\r+/m)[1];
      let insert = txt.match(/\r+(.*)$/m)[1];
      insert = insert + base.slice(insert.length, base.length);
      txt = txt.replace(/\r+.*$/m, '\r').replace(/^.*\r/m, insert);
    }
    return txt;
  }
```

So recommendation is
```ts
    txt = txt.replace(/\r+\n/gm, '\n'); // \r followed by \n --> newline